include/SDL3/SDL_gpu.h

```diff
@@ -0,0 +1,2263 @@
+    SDL_GPU_LOADOP_DONT_CARE
+} SDL_GpuLoadOp;
+
+typedef enum SDL_GpuStoreOp
+{
+    SDL_GPU_STOREOP_STORE,
+    SDL_GPU_STOREOP_DONT_CARE
+} SDL_GpuStoreOp;
+
+typedef enum SDL_GpuIndexElementSize
+{
+    SDL_GPU_INDEXELEMENTSIZE_16BIT,
+    SDL_GPU_INDEXELEMENTSIZE_32BIT
+} SDL_GpuIndexElementSize;
+
+typedef enum SDL_GpuTextureFormat
+{
+    SDL_GPU_TEXTUREFORMAT_INVALID = -1,
+
+    /* Unsigned Normalized Float Color Formats */
+    SDL_GPU_TEXTUREFORMAT_R8G8B8A8,
+    SDL_GPU_TEXTUREFORMAT_B8G8R8A8,
+    SDL_GPU_TEXTUREFORMAT_B5G6R5,
+    SDL_GPU_TEXTUREFORMAT_B5G5R5A1,
+    SDL_GPU_TEXTUREFORMAT_B4G4R4A4,
+    SDL_GPU_TEXTUREFORMAT_R10G10B10A2,
+    SDL_GPU_TEXTUREFORMAT_R16G16,
+    SDL_GPU_TEXTUREFORMAT_R16G16B16A16,
+    SDL_GPU_TEXTUREFORMAT_R8,
+    SDL_GPU_TEXTUREFORMAT_A8,
+    /* Compressed Unsigned Normalized Float Color Formats */
+    SDL_GPU_TEXTUREFORMAT_BC1,
+    SDL_GPU_TEXTUREFORMAT_BC2,
+    SDL_GPU_TEXTUREFORMAT_BC3,
+    SDL_GPU_TEXTUREFORMAT_BC7,
+    /* Signed Normalized Float Color Formats  */
+    SDL_GPU_TEXTUREFORMAT_R8G8_SNORM,
+    SDL_GPU_TEXTUREFORMAT_R8G8B8A8_SNORM,
+    /* Signed Float Color Formats */
+    SDL_GPU_TEXTUREFORMAT_R16_SFLOAT,
+    SDL_GPU_TEXTUREFORMAT_R16G16_SFLOAT,
+    SDL_GPU_TEXTUREFORMAT_R16G16B16A16_SFLOAT,
+    SDL_GPU_TEXTUREFORMAT_R32_SFLOAT,
+    SDL_GPU_TEXTUREFORMAT_R32G32_SFLOAT,
+    SDL_GPU_TEXTUREFORMAT_R32G32B32A32_SFLOAT,
+    /* Unsigned Integer Color Formats */
+    SDL_GPU_TEXTUREFORMAT_R8_UINT,
+    SDL_GPU_TEXTUREFORMAT_R8G8_UINT,
+    SDL_GPU_TEXTUREFORMAT_R8G8B8A8_UINT,
+    SDL_GPU_TEXTUREFORMAT_R16_UINT,
+    SDL_GPU_TEXTUREFORMAT_R16G16_UINT,
+    SDL_GPU_TEXTUREFORMAT_R16G16B16A16_UINT,
+    /* SRGB Color Formats */
+    SDL_GPU_TEXTUREFORMAT_R8G8B8A8_SRGB,
+    SDL_GPU_TEXTUREFORMAT_B8G8R8A8_SRGB,
+    /* Compressed SRGB Color Formats */
+    SDL_GPU_TEXTUREFORMAT_BC3_SRGB,
+    SDL_GPU_TEXTUREFORMAT_BC7_SRGB,
+    /* Depth Formats */
+    SDL_GPU_TEXTUREFORMAT_D16_UNORM,
+    SDL_GPU_TEXTUREFORMAT_D24_UNORM,
+    SDL_GPU_TEXTUREFORMAT_D32_SFLOAT,
+    SDL_GPU_TEXTUREFORMAT_D24_UNORM_S8_UINT,
+    SDL_GPU_TEXTUREFORMAT_D32_SFLOAT_S8_UINT
+} SDL_GpuTextureFormat;
+
+typedef enum SDL_GpuTextureUsageFlagBits
+{
+    SDL_GPU_TEXTUREUSAGE_SAMPLER_BIT = 0x00000001,
+    SDL_GPU_TEXTUREUSAGE_COLOR_TARGET_BIT = 0x00000002,
+    SDL_GPU_TEXTUREUSAGE_DEPTH_STENCIL_TARGET_BIT = 0x00000004,
+    SDL_GPU_TEXTUREUSAGE_GRAPHICS_STORAGE_READ_BIT = 0x00000008,
+    SDL_GPU_TEXTUREUSAGE_COMPUTE_STORAGE_READ_BIT = 0x00000020,
+    SDL_GPU_TEXTUREUSAGE_COMPUTE_STORAGE_WRITE_BIT = 0x00000040
+} SDL_GpuTextureUsageFlagBits;
+
+typedef Uint32 SDL_GpuTextureUsageFlags;
+
+typedef enum SDL_GpuTextureType
+{
+    SDL_GPU_TEXTURETYPE_2D,
+    SDL_GPU_TEXTURETYPE_3D,
+    SDL_GPU_TEXTURETYPE_CUBE,
+} SDL_GpuTextureType;
+
+typedef enum SDL_GpuSampleCount
+{
+    SDL_GPU_SAMPLECOUNT_1,
+    SDL_GPU_SAMPLECOUNT_2,
+    SDL_GPU_SAMPLECOUNT_4,
+    SDL_GPU_SAMPLECOUNT_8
```

Comment:
No 16x MSAA?

Reply:
I think Ryan wanted this to be an int anyway, so let's just scrap the enum altogether.

Reply:
Is there a task tracking this?